src/sentry/search/events/fields.py

```diff
@@ -2408,10 +2475,16 @@ def resolve_function(self, function: str, match: Optional[Match[str]] = None) ->
         snql_function = self.function_converter.get(name)
 
         arguments = snql_function.format_as_arguments(name, arguments, self.params)
+        for arg in snql_function.args:
+            if type(arg) == SnQLColumn:
+                arguments[arg.name] = self.column(arguments[arg.name])
```

Comment:
I like the direction this is moving in. A few nitpicks though:
1. should this use `self.resolve_field` instead to support field aliases, not necessary for `to_other` but will be necessary soon
2. would it make sense to spin off a custom implementation of `format_as_arguments` for `SnQLFunction`? It is still doing the work of resolving the name to the json syntax, only to be overwritten after the fact.

Reply:
1. I think since this arg type is specifically for Column type, it makes sense to resolve it with self.column? If we introduce a more generic FunctionArg class that could be a column or field alias, then it makes sense to have an elif statement that uses `self.resolve_field`?
2. I mean with the snql specific subclassing we are essentially only validating and not really resolving stuff to the JSON syntax anymore. I think in either case, the column/field has to be resolved in here, so I'm not sure if it's necessary

Reply:
Ok lol I changed my mind on 1. Since all the validation should happen in the normalize method, using resolve fields should be fine